strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/storage/repository/RepositoryPolicyEnum.java

```diff
@@ -1,14 +1,30 @@
 package org.carlspring.strongbox.storage.repository;
 
+import java.util.stream.Stream;
+
 /**
  * @author mtodorov
  */
 public enum RepositoryPolicyEnum
 {
 
-    RELEASE("release"),
+    RELEASE("release")
+            {
+                @Override
+                public boolean acceptsSnapshots()
+                {
+                    return false;
+                }
+            },
 
-    SNAPSHOT("snapshot"),
+    SNAPSHOT("snapshot")
+            {
+                @Override
+                public boolean acceptsReleases()
+                {
+                    return false;
+                }
+            },
 
     MIXED("mixed");
```

Comment:
I'd recommend dropping the default `true` boolean values for the two fields below and having an explicit override of both methods. Makes it immediately clear when looking at the code, although it's still the same in effect.